### DISCUSSION TITLE
A right way to store object position?

### DISCUSSION THREAD

**Discussion by sqrvrt** [bevyengine/bevy#9358] [Q&A]
I'm currently making a project where i need to store objects with `SpriteBundle`s . However storing position and rotation directly in spritebundles is inconvenient for me (as i need to multiply rotation by pi for example). Can i configure spritebundles to source position from somewhere else or do i think entirely wrong about ECS?

**Comment by nicopap** [comment]
What are you actually trying to do?

I think you might have some misconception about the ECS. What do you mean by "storing in sprite bundle"? Bundles only exist for a very short time. Once the components are written to the ECS, the bundle doesn't exist anymore. You have an `Entity` with several components (`Transform`, `GlobalTransform`, `Sprite`, `Handle<Image>` etc.)

What do you intend to do when you say "i need to multiply rotation by pi"?

What do you mean "configure spritebundles to source position from somewhere else"? I don't understand. Could you precise what you want to do?

**Reply by rlidwka** [reply]
Perhaps, he needs to have his Entity (as game object) to have one transform, but visuals (sprite) to have that transform modified (because the asset he has is rotated in different way).

If that's the case, I think sprite has to be put in a separate entity as a child of game object, not in the same entity. I'm running into this issue quite frequently.

**Reply by rlidwka** [reply]
And "*configure spritebundles to source position from somewhere else*" can mean "load sprite/scene asset with modified transform", i.e. physically rotate everything inside it when it loads before inserting into entity. Don't think that's possible.

**Reply by sqrvrt** [reply]
I think i misunderstood how the ecs works but what i ideally want now is some kind of a wrapper of data, so say the rotation is stored as 3 the value it gets rendered as will be (pi / 2) * 3.

**Reply by rlidwka** [reply]
> I think i misunderstood how the ecs works but what i ideally want now is some kind of a wrapper of data, so say the rotation is stored as 3 the value it gets rendered as will be (pi / 2) * 3.

If you want that, you can add a custom component that stores `3`, and a system that would sync the value between your component and Transform.

**Comment by sqrvrt** [comment]
I think I pretty much got the basis now, but there's still the initialization question. Even tho components are stored just on entities, when you init you still need to write something like
```rust
let test = testbundle {
    something: SomeComponent,
    display: SpriteBundle {
        transform: Transform::from_translation(...).with_scale(...),
        ..default();
    }
}
```
which works but looks odd cuz there are properties in the spritebundle which are purely display-related (e.g. texture) and there is stuff that actually matters, like transform.
